exporter/elasticsearchexporter/model.go

```diff
@@ -163,6 +169,125 @@ func (m *encodeModel) encodeLogECSMode(resource pcommon.Resource, record plog.Lo
 	return document
 }
 
+func (m *encodeModel) encodeMetrics(resource pcommon.Resource, metrics pmetric.MetricSlice, _ pcommon.InstrumentationScope) ([][]byte, error) {
+	var baseDoc objmodel.Document
+
+	baseDoc.AddAttributes("", resource.Attributes())
+
+	// Put all metrics that have the same attributes and timestamp in one document.
+	docs := map[uint32]*objmodel.Document{}
+	for i := 0; i < metrics.Len(); i++ {
+		metric := metrics.At(i)
+
+		var dps pmetric.NumberDataPointSlice
+
+		// Only Gauge and Sum metric types are supported at the moment.
+		switch metric.Type() {
+		case pmetric.MetricTypeGauge:
+			dps = metric.Gauge().DataPoints()
+		case pmetric.MetricTypeSum:
+			dps = metric.Sum().DataPoints()
+		}
+
+		for j := 0; j < dps.Len(); j++ {
+			dp := dps.At(j)
+
+			hash := metricHash(dp.Timestamp(), dp.Attributes())
+			doc, docExists := docs[hash]
```

Comment:
non-blocker: Are we worried about hash collisions? In apm-data it doesn't just compare hashes, but compares string-based map keys, which in go implementation should first compare hash then actual bytes (see [code](https://github.com/elastic/apm-data/blob/main/input/otlp/metrics.go#L378)). However, at the same time, apm-data is vulnerable to value->string collisions.

Reply:
Oh this whole hashing detail will actually be removed in a follow up PR ([draft here](https://github.com/andrzej-stencel/opentelemetry-collector-contrib/pull/1)) due to a change in how data is routed. Sorry for the trouble, probably no need reviewing this in light of the upcoming change.